crypto/s2n_drbg.c

```diff
@@ -136,10 +136,28 @@ int s2n_drbg_instantiate(struct s2n_drbg *drbg, struct s2n_blob *personalization
     return 0;
 }
 
+int s2n_new_aes128_drbg(struct s2n_drbg *drbg, struct s2n_blob *personalization_string)
+{
+    drbg->ctx = EVP_CIPHER_CTX_new();
+    S2N_ERROR_IF(!drbg->ctx, S2N_ERR_DRBG);
+    (void)EVP_CIPHER_CTX_init(drbg->ctx);
```

Comment:
Please ignore this comment if S2N is used with OpenSSL 1.0.2 (https://www.openssl.org/docs/man1.0.2/man3/EVP_EncryptInit.html)

However, for S2N with OpenSSL 1.1.0 note that in: https://www.openssl.org/docs/man1.1.1/man3/EVP_EncryptInit_ex.html
 
> EVP_CIPHER_CTX was made opaque in OpenSSL 1.1.0. As a result, EVP_CIPHER_CTX_reset() appeared and EVP_CIPHER_CTX_cleanup() disappeared. EVP_CIPHER_CTX_init() remains as an alias for EVP_CIPHER_CTX_reset().

and that:
> EVP_CIPHER_CTX_reset() returns 1 for success and 0 for failure.
So, it can fail, should the return value be ignored here?

Reply:
https://github.com/openssl/openssl/blob/master/include/openssl/opensslv.h has undergone a lot of changes and newer versions have a nice `OPENSSL_VERSION_PREREQ`, but back in 1.0.X they don't so I added a check for 0x10100000L aka 1.1.0

Reply:
I found s2n has a nice helper method for this already.